pkg/yurtctl/cmd/clusterinfo/clusterinfo.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2020 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterinfo
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	ct "github.com/daviddengcn/go-colortext"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	"github.com/alibaba/openyurt/pkg/projectinfo"
+	kubeutil "github.com/alibaba/openyurt/pkg/yurtctl/util/kubernetes"
+
+)
+
+
+
+// ClusterInfoOptions has the information that required by cluster-info operation
+type ClusterInfoOptions struct {
+	clientSet             *kubernetes.Clientset
+	CloudNodes            []string
+	EdgeNodes             []string
+}
+
+// NewClusterInfoOptions creates a new ClusterInfoOptions
+func NewClusterInfoOptions() *ClusterInfoOptions {
+	return &ClusterInfoOptions{
+		CloudNodes: []string{},
+	}
+}
+
+// NewClusterInfoCmd generates a new cluster-info command
+func NewClusterInfoCmd() *cobra.Command {
+	o := NewClusterInfoOptions()
+	cmd := &cobra.Command{
+		Use:   "cluster-info",
+		Short: "show cluster info",
+		Run: func(cmd *cobra.Command, _ []string) {
+			if err := o.Complete(cmd.Flags()); err != nil {
+				klog.Fatalf("fail to complete the cluster-info option: %s", err)
+			}
+			if err := o.Run(); err != nil {
+				klog.Fatalf("fail to run cluster-info cmd: %s", err)
+			}
+		},
+	}
+
+	return cmd
+}
+
+// Complete completes all the required options
+func (o *ClusterInfoOptions) Complete(flags *pflag.FlagSet) error {
+	var err error
+	o.clientSet, err = kubeutil.GenClientSet(flags)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Validate makes sure provided values for ClusterInfoOptions are valid
+func (o *ClusterInfoOptions) Validate() error {
+	return nil
+}
+
+func (o *ClusterInfoOptions) Run() (err error) {
+	// 1. list edge node
+	label := fmt.Sprintf("%s=true", projectinfo.GetEdgeWorkerLabelKey())
+	edgeNodes, err := o.clientSet.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: label})
```

Comment:
now twice list request to get cloud nodes and edge nodes, how about only one list request to get all nodes, and then extract the cloud nodes and edge nodes? because one list request can be reduced.

Reply:
That sounds great. I will change them. Thanks.